oshmem/mca/memheap/base/memheap_base_static.c

```diff
@@ -127,7 +130,47 @@ static int _check_address(struct map_segment_desc *seg)
 
 static int _check_pathname(struct map_segment_desc *seg)
 {
-    /* Probably we need to check found path but
+    static volatile int init = 0;
+    static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+    static const char *proc_self_exe = "/proc/self/exe";
+    static char exe_path[PATH_MAX] = "";
+    char module_path[PATH_MAX];
+    char *path;
+
+    if (0 == seg->inode) {
+        /* segment is not mapped to file, allow sharing */
+        return OSHMEM_SUCCESS;
+    }
+
+    if (0 == init) {
+        pthread_mutex_lock(&mutex);
+        if (0 == init) {
+            path = realpath(proc_self_exe, exe_path);
```

Comment:
imo, better not cache it: it increases binary size and almost does not improve perf. we call realpath once anyway for each seg.

Reply:
ok, caching is removed